### ISSUE TITLE
[Question] Using different private keys in ECIES_Encryptor, does it provide more security?

### ISSUE THREAD

**Issue by CingKong** [randombit/botan#2299]
If same parameters (including a private key for constructing the encryptor) are passed to ECIES_Encryptor, it gives same output, because ECIES_KA_Operation::derive_secret() generates the exact same secret.

And so it makes me curious, does generating different private keys for ECIES_Encryptor improve security? If so and if I randomize IV every time before encryption, does constructing an encryptor with a different private key still make sense?

**Comment by CingKong** [comment]
And an unrelated question,
in [ECIES_Encryptor::maximum_input_size()](https://github.com/randombit/botan/blob/06a55af16fbee536cf3aa53370195b22327ca54f/src/lib/pubkey/ecies/ecies.cpp#L259) the comment says "ECIES should just be used for key transport".

why is that?

**Comment by mouse07410** [comment]
Something's wrong here. *Note that I'm talking about ECIES in general, rather than it's Botan implementation.*

First, ECIES is a hybrid encryption system designed to secure (reasonably) large data files. It derives symmetric keys from the recipient public key and the sender ephemeral (random in practice) private key, encrypts the data and generated authentication tag, and packs the encrypted data, the tag, and the sender's ephemeral public key into a blob that gets sent to the recipient.

I do recall that Botan was doing something strange with ECIES, like passing **private** key to the **Encryptor** (instead of **De**cryptor), but I don't remember what that was about.

In summary: a normal ECIES implementation **certainly** can be used for securing data, but just the keys.

**Comment by CingKong** [comment]
You has mentioned about the constructor taking a private key in [this issue](https://github.com/randombit/botan/issues/2150#issuecomment-541434488), in there randombit said because private keys can be converted to public keys… 

Though all I got from reading Botan's source code is, in encryptor some where it calls `Private_Key::create_key_agreement_op()`, and so it does need a private key. :P

**Comment by mouse07410** [comment]
Key agreement needs an *ephemeral* private key for the sender/encryptor.

Ok, I think I've contributed here as much as I could, and probably more than I should.

**Comment by CingKong** [comment]
> That key is supposed to be generated _randomly_ by the encryptor itself.

Botan does have a [constructor](https://github.com/randombit/botan/blob/5d74496ee51b8a2d1c418b0a66bddac6f0263749/src/lib/pubkey/ecies/ecies.cpp#L254) for it that generates a new private key instead of taking an existing one.

Here is my original question, rephrased: Do I, for better security, need to use a new and different encryptor to encrypt data every time, does it matter (even if I randomize IV)?

**Comment by CingKong** [comment]
Not sure if it's Botan's implementation only or ECIES  is written this way (I don't read papers), (at least in Botan) the key used for `Cipher_Mode` and `MessageAuthenticationCode` can be derived using **one party's private key and the other one's public key**, it doesn't matter which one is the sender, which is the receiver.

And so in practice one shouldn't keep the private key used for ECIES at all?
